### PR TITLE
Support more placeholder types

### PR DESCRIPTION
**Pull request by iampatbrown**
Thoughts on `_generatePlaceholder()` supporting additional types?

I haven't thought through this too much, just wanted to run it by you.

### CODE REVIEW

File: Sources/XCTestDynamicOverlay/XCTUnimplemented.swift [hunk 1]
```diff
@@ -379,12 +397,68 @@ private func _unimplementedFatalError(_ message: String, file: StaticString, lin
...
+}
+extension Bool: DefaultConstructible {
+  static var defaultValue: Self { false }
+}
+extension Set: DefaultConstructible {
+  static var defaultValue: Self { [] }
+}
+extension Dictionary: DefaultConstructible {
+  static var defaultValue: Self { [:] }
+}
```
Comment (iampatbrown) [inline Sources/XCTestDynamicOverlay/XCTUnimplemented.swift:453]: These are a bit redundant as they are cover by `AnyExpressibleByArrayLiteral` and `AnyExpressibleByDictionaryLiteral`

File: Sources/XCTestDynamicOverlay/XCTUnimplemented.swift [hunk 2]
```diff
@@ -379,12 +394,58 @@ private func _unimplementedFatalError(_ message: String, file: StaticString, lin
...
+    Value.zero
+  }
+}
+
+protocol DefaultConstructible {
+  init()
+}
+extension Bool: DefaultConstructible {}
+extension Set: DefaultConstructible {}
+extension Dictionary: DefaultConstructible {}
```
Comment (iampatbrown) [inline Sources/XCTestDynamicOverlay/XCTUnimplemented.swift:444]: These are a bit redundant as they are cover by `AnyExpressibleByArrayLiteral` and `AnyExpressibleByDictionaryLiteral`

**Comment by stephencelis** [comment]
@iampatbrown Sorry for the delay. We're pretty busy at the moment, but this definitely looks interesting! Might be a week or two before we have time to think through the current implementation, and do feel free to ping us here if we neglect it for too long!

**Comment by iampatbrown** [comment]
> Might be a week or two before we have time to think through the current implementation

@stephencelis No stress at all. Figured the implementation might need some thought :)

**Comment by iampatbrown** [comment]
Closing this for now. The current works for the majority of cases and using an explicit placeholder when needed is fine.

**Comment by stephencelis** [comment]
@iampatbrown Down to revisit this? We've moved things around, so if you don't have time I can attempt to rebase, but also wanted to make sure there wasn't anything else that came to mind since you first opened this.

**Comment by iampatbrown** [comment]
@stephencelis I'd most likely redo with a 5.7 implementation as well. Can resist this on the weekend if there's no rush.

**Pull request closed without merging**